test/Ocelot.AcceptanceTests/Steps.cs

```diff
@@ -566,38 +644,38 @@ public void GivenOcelotIsRunningWithGlobalHandlersRegisteredInDi<TOne>(FakeDepen
                 s.AddOcelot()
                     .AddDelegatingHandler<TOne>(true);
             })
-            .Configure(a => { a.UseOcelot().Wait(); });
+            .Configure(async app => { await app.UseOcelot(); });
 
         _ocelotServer = new TestServer(_webHostBuilder);
 
         _ocelotClient = _ocelotServer.CreateClient();
-    }
-
-    // #
-    // # Cookies helpers
-    // #
-    public void GivenIAddCookieToMyRequest(string cookie)
+    }
+
+    // #
+    // # Cookies helpers
+    // #
+    public void GivenIAddCookieToMyRequest(string cookie)
         => _ocelotClient.DefaultRequestHeaders.Add("Set-Cookie", cookie);
-    public async Task WhenIGetUrlOnTheApiGatewayWithCookie(string url, string cookie, string value)
-        => _response = await WhenIGetUrlOnTheApiGateway(url, cookie, value);
-    public async Task WhenIGetUrlOnTheApiGatewayWithCookie(string url, CookieHeaderValue cookie)
-        => _response = await WhenIGetUrlOnTheApiGateway(url, cookie);
-
-    public Task<HttpResponseMessage> WhenIGetUrlOnTheApiGateway(string url, string cookie, string value)
-    {
-        var header = new CookieHeaderValue(cookie, value);
-        return WhenIGetUrlOnTheApiGateway(url, header);
-    }
-
-    public Task<HttpResponseMessage> WhenIGetUrlOnTheApiGateway(string url, CookieHeaderValue cookie)
-    {
-        var requestMessage = new HttpRequestMessage(HttpMethod.Get, url);
-        requestMessage.Headers.Add("Cookie", cookie.ToString());
-        return _ocelotClient.SendAsync(requestMessage);
-    }
-
+    public async Task WhenIGetUrlOnTheApiGatewayWithCookieAsync(string url, string cookie, string value)
+        => _response = await WhenIGetUrlOnTheApiGatewayAsync(url, cookie, value);
+    public async Task WhenIGetUrlOnTheApiGatewayWithCookieAsync(string url, CookieHeaderValue cookie)
+        => _response = await WhenIGetUrlOnTheApiGatewayAsync(url, cookie);
+
+    public async Task<HttpResponseMessage> WhenIGetUrlOnTheApiGatewayAsync(string url, string cookie, string value)
+    {
+        var header = new CookieHeaderValue(cookie, value);
+        return await WhenIGetUrlOnTheApiGatewayAsync(url, header);
```

Comment:
Why do you await here? 🙄

Reply:
I realize I misunderstand this part at the async guide at [Prefer async/await over directly returning Task](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task). I agree we should just return the `Task`. 

Reading the piece below... Is it looks confuse for you too?

![image](https://github.com/user-attachments/assets/5b70ed31-7c2e-4a8e-850f-54ff1327fb29)

Reply:
I can't dispute Mr. Fowler's point, and his advice is indeed beneficial for the library code, not the testing project. Performance is my priority during testing: tests need to be quick. I'm not concerned about leaking internal variables into the broader context in testing projects or a disrupted call stack.

Please fix this!

Reply:
Already did!